test/extended/marketplace/source_create.go

```diff
@@ -0,0 +1,163 @@
+package marketplace
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:Marketplace] Marketplace basic", func() {
+
+	defer g.GinkgoRecover()
+
+	var (
+		oc            = exutil.NewCLI("marketplace", exutil.KubeConfigPath())
+		allNs         = "openshift-operators"
+		marketplaceNs = "openshift-marketplace"
+
+		//buildPruningBaseDir = exutil.FixturePath("testdata", "marketplace")
+		opsrcYamltem = exutil.FixturePath("testdata", "marketplace", "opsrc", "01-opsrc.yaml")
+		cscYamltem   = exutil.FixturePath("testdata", "marketplace", "csc", "01-csc.yaml")
+		subYamltem   = exutil.FixturePath("testdata", "marketplace", "sub", "01-subofdes.yaml")
+	)
+
+	g.AfterEach(func() {
+		//clear the sub,csv resource
+		allresourcelist := [][]string{
+			{"operatorsource", "opsrctest", marketplaceNs},
+			{"catalogsourceconfig", "csctest", marketplaceNs},
+			{"csv", "descheduler.v0.0.3", allNs},
+			{"subscription", "descheduler-test", allNs},
+		}
+		for _, source := range allresourcelist {
+			err := clearResources(oc, source[0], source[1], source[2])
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+
+	//OCP-21405 create one opsrc,OCP-21419 create one csc,OCP-21479 sub to openshift-operators,OCP-21667 delete the csc
+	g.It("[ocp-21405 21419 21479 21667]create and delete the basic source,sub one operator", func() {
+
+		//create one opsrc
+		//opsrcYaml := exutil.FixturePath("testdata", "marketplace", "opsrc", "01-opsrc.yaml")
+		opsrcYaml, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", opsrcYamltem, "-p", "NAME=opsrctest", "NAMESPACE=jfan", fmt.Sprintf("MARKETPLACE=%s", marketplaceNs)).OutputToFile("config.json")
```

Comment:
Is it a good idea to use this namespace? What happens if it gets deleted or needs to be updated for some reason? Can we create a namespace that the whole team has access to?

Marketplace has a similar e2e test that uses a common namespace, maybe we could create something similar or even share it across these tests?

Reply:
Hmm a good idea. may I get the access key to the public ns ?

Reply:
We use the "marketplace_e2e" namespace in our tests: https://github.com/operator-framework/operator-marketplace/blob/master/test/helpers/helpers.go#L267

It is a public namespace so there is no secret required, you should be able to access it.

Reply:
You mean the ns  "marketplace_e2e"  ? But I have to push my test csv "descheduler-test" to it. Could you give me the token?